editor/components/inserter-with-shortcuts/index.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+import { filter } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { BlockIcon, createBlock, getDefaultBlockName } from '@wordpress/blocks';
+import { compose } from '@wordpress/element';
+import { IconButton, withContext } from '@wordpress/components';
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import Inserter from '../inserter';
+import { getMostFrequentlyUsedBlocks } from '../../store/selectors';
+import { replaceBlocks } from '../../store/actions';
+
+function InserterWithShortcuts( { items, isLocked, onToggle, onInsert } ) {
+	if ( isLocked ) {
+		return null;
+	}
+
+	const itemsWithTheDefaultBlock =
+		filter( items, ( item ) => item.name !== getDefaultBlockName() || ! item.initialAttributes )
```

Comment:
`! item.initialAttributes` will _always_ evaluate to false. I think this should be:

```js
const itemsWithoutDefaultBlock = filter( items, ( item ) =>
    item.name !== getDefaultBlockName() || Object.keys( item.initialAttributes ).length > 0
).slice( 0, 2 );
```

Using `without` and `isEmpty` would help make it read more naturally:

```js
const itemsWithoutDefaultBlock = without( items, ( item ) =>
    item.name === getDefaultBlockName() && isEmpty( item.initialAttributes )
).slice( 0, 2 );
```